### PR TITLE
Composite VS Components Telemetry

### PR DESCRIPTION
**Pull request by palatter**
# What

# Why

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/acs-ui-common/review/beta/acs-ui-common.api.md [hunk 1]
```diff
@@ -85,6 +85,11 @@ export type TelemetryEvent = {
     data?: Record<string, unknown>;
 };
 
+// Warning: (ae-internal-missing-underscore) The name "TelemetryImplementationHint" should be prefixed with an underscore because the declaration is marked as @internal
```
Comment (edwardlee-msft) [inline packages/acs-ui-common/review/beta/acs-ui-common.api.md:88]: internal type needs to be prefixed with an _ (underscore)

File: packages/calling-stateful-client/review/beta/calling-stateful-client.api.md [hunk 2]
```diff
@@ -168,6 +169,11 @@ export interface CaptionsInfo {
 // @public
 export const createStatefulCallClient: (args: StatefulCallClientArgs, options?: StatefulCallClientOptions | undefined) => StatefulCallClient;
 
+// Warning: (ae-internal-missing-underscore) The name "createStatefulCallClientInner" should be prefixed with an underscore because the declaration is marked as @internal
```
Comment (edwardlee-msft) [inline packages/calling-stateful-client/review/beta/calling-stateful-client.api.md:172]: Internal code should be prefixed with an underscore.
Can we please add a short comment describing these internal functions as well for others.

File: packages/chat-stateful-client/review/beta/chat-stateful-client.api.md [hunk 3]
```diff
@@ -71,6 +72,11 @@ export type ChatThreadProperties = {
 // @public
 export const createStatefulChatClient: (args: StatefulChatClientArgs, options?: StatefulChatClientOptions | undefined) => StatefulChatClient;
 
+// Warning: (ae-internal-missing-underscore) The name "createStatefulChatClientInner" should be prefixed with an underscore because the declaration is marked as @internal
```
Comment (edwardlee-msft) [inline packages/chat-stateful-client/review/beta/chat-stateful-client.api.md:75]: internal, prefix with an underscore here

File: packages/react-composites/src/composites/ChatComposite/adapter/AzureCommunicationChatAdapter.ts [hunk 4]
```diff
@@ -518,16 +518,36 @@ export const createAzureCommunicationChatAdapter = async ({
...
-    userId,
-    displayName,
-    endpoint: endpointUrl,
-    credential: credential
-  });
+  const chatClient = _createStatefulChatClientInner(
+    {
+      userId,
+      displayName,
+      endpoint: endpoint,
```
Comment (alkwa-msft) [inline packages/react-composites/src/composites/ChatComposite/adapter/AzureCommunicationChatAdapter.ts:545]: if the  values match the key then we can just say 

{ userId, displayName, endpoint, credential }

File: packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts [hunk 5]
```diff
@@ -1407,9 +1436,13 @@ export const createTeamsCallAdapter = async ({
...
 }: TeamsCallAdapterArgs): Promise<TeamsCallAdapter> => {
-  const callClient = createStatefulCallClient({
-    userId
-  });
+  const callClient = _createStatefulCallClientInner(
+    {
+      userId
+    },
+    undefined,
+    'Call' as _TelemetryImplementationHint
```
Comment (alkwa-msft) [inline packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts:1444]: can we review this with internal devx group?

File: packages/acs-ui-common/review/beta/acs-ui-common.api.md [hunk 6]
```diff
@@ -85,6 +85,9 @@ export type TelemetryEvent = {
     data?: Record<string, unknown>;
 };
 
+// @internal
+export type _TelemetryImplementationHint = 'Call' | 'Chat' | 'CallWithChat' | 'StatefulComponents';
```
Comment (alkwa-msft) [inline packages/acs-ui-common/review/beta/acs-ui-common.api.md:89]: can we review if we want to use these enums with the team?

I am on the fence about StatefulComponents but im not sure what we should be doing instead.

CallComposite, ChatComposite, CallWithChatComposite, and then default for components?

**Review by alkwa-msft — APPROVED** [review]
It looks like it works but id like to have this design reviewed across the UI team and possibly internal devx folks as well

**Review by edwardlee-msft — APPROVED** [review]

**Review by JamesBurnside — APPROVED** [review]

File: packages/acs-ui-common/src/telemetry.ts [hunk 7]
```diff
@@ -31,7 +58,10 @@ export const sanitize = (version: string): string => {
  *
  * @internal
  */
-export const _getApplicationId = (): string => {
+export const _getApplicationId = (telemetryImplementationHint?: _TelemetryImplementationHint): string => {
```
Comment (JamesBurnside) [inline packages/acs-ui-common/src/telemetry.ts:61]: can we make the param here required:
```suggestion
export const _getApplicationId = (telemetryImplementationHint: _TelemetryImplementationHint): string => {
```

Same as `getTelemetryImplementationHint`, ideally we should always be passing a hint into them

**Review by JoshuaLai — APPROVED** [review]

**Pull request merged**